### PR TITLE
HEAD option for sphinxbase and pocketsphinx

### PR DESCRIPTION
**Pull request by turbolent**

### CODE REVIEW

File: Library/Formula/cmu-pocketsphinx.rb [hunk 1]
```diff
@@ -4,13 +4,21 @@ class CmuPocketsphinx < Formula
...
   sha1 'd9efdd0baddd2e47c2ba559caaca62ffa0c0eede'
+  head "https://github.com/cmusphinx/pocketsphinx.git"
 
   depends_on 'pkg-config' => :build
+  if build.head?
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "libtool" => :build
+    depends_on "swig" => :build
+  end
```
Comment (jacknagel) [inline Library/Formula/cmu-pocketsphinx.rb]: `head` dependencies should be in a block, along with the url, like so:

```rb
head do
  url "..."

  depends_on "..."
  depends_on "..."
end
```

File: Library/Formula/cmu-pocketsphinx.rb [hunk 2]
```diff
@@ -4,13 +4,21 @@ class CmuPocketsphinx < Formula
...
   depends_on 'cmu-sphinxbase'
 
   def install
-    system "./configure", "--disable-dependency-tracking",
-                          "--prefix=#{prefix}"
+    system (if build.head? then "./autogen.sh" else "./configure" end),
+      "--disable-dependency-tracking",
+      "--prefix=#{prefix}"
```
Comment (jacknagel) [inline Library/Formula/cmu-pocketsphinx.rb]: Looks like one can do:

```rb
if build.head?
  ENV["NONCOFIGURE"] = "1"
  system "./autogen.sh"
end
```

and leave the rest as-is.

**Comment by turbolent** [comment]
Thanks for the feedback! Now using head blocks and the `NOCONFIGURE` environment variable.

File: Library/Formula/cmu-pocketsphinx.rb [hunk 3]
```diff
@@ -5,10 +5,23 @@ class CmuPocketsphinx < Formula
...
+    depends_on "automake" => :build
+    depends_on "libtool" => :build
+    depends_on "swig" => :build
+  end
+
   depends_on 'pkg-config' => :build
   depends_on 'cmu-sphinxbase'
 
   def install
+    if build.head? then
```
Comment (mikemcquaid) [inline Library/Formula/cmu-pocketsphinx.rb]: don't need the `then`

File: Library/Formula/cmu-sphinxbase.rb [hunk 4]
```diff
@@ -12,6 +21,10 @@ class CmuSphinxbase < Formula
   depends_on "libsamplerate" => "with-libsndfile"
 
   def install
+    if build.head? then
```
Comment (mikemcquaid) [inline Library/Formula/cmu-sphinxbase.rb]: Don't need the `then`

**Pull request closed without merging**